syntax = "proto2";

package Authentication;

enum reason {
  ACCOUNT_ALREADY_EXISTS = 0;
  INVALID_CREDENTIALS = 1;
  INTERNAL_SERVER_ERROR = 2;
}

message CreateAccountPacket {
  required int32 userId = 1;
  required string email = 2;
  required string salt = 3;
  required string hashed_password = 4;
}

message CreateAccountSuccessPacket {
  optional int32 userId = 1;
}

message CreateAccountFailurePacket {
  optional int32 userId = 1;  
  required reason reasonId = 2;
}

message LoginPacket {
  optional int32 userId = 1;
  required string email = 2;
  required string hashed_password = 3;
}

message LoginSuccessPacket {
  optional int32 userId = 1;
}

message LoginFailurePacket {
  optional int32 userId = 1;  
  required reason reasonId = 2;
}
